service/internal/access/v2/validators.go

```diff
@@ -0,0 +1,140 @@
+)
+
+var (
+	ErrInvalidAction                = errors.New("access: invalid action")
+	ErrInvalidEntityChain           = errors.New("access: invalid entity chain")
+	ErrInvalidEntitledFQNsToActions = errors.New("access: invalid entitled FQNs to actions")
+)
+
+// validateGetDecision validates the input parameters for GetDecision:
+//
+//   - entityRepresentation: must not be nil
+//   - action: must not be nil
+//   - resources: must not be nil and must contain at least one resource
+func validateGetDecision(entityRepresentation *entityresolutionV2.EntityRepresentation, action *policy.Action, resources []*authzV2.Resource) error {
+	if err := validateEntityRepresentations([]*entityresolutionV2.EntityRepresentation{entityRepresentation}); err != nil {
+		return fmt.Errorf("invalid entity representation: %w", err)
+	}
+	if action.GetName() == "" {
+		return fmt.Errorf("action required with name: %w", ErrInvalidAction)
+	}
+	if len(resources) == 0 {
+		return fmt.Errorf("resources are empty: %w", ErrInvalidResource)
+	}
+	for _, resource := range resources {
+		if resource == nil {
+			return fmt.Errorf("resource is nil: %w", ErrInvalidResource)
+		}
+	}
+	return nil
+}
+
+// validateSubjectMapping validates the subject mapping is valid for an entitlement decision
+//
+// subjectMapping:
+//
+//   - must not be nil
+//   - must have a non-empty attribute value
+//   - must have a non-empty attribute value FQN
+//   - must have a non-empty actions
+func validateSubjectMapping(subjectMapping *policy.SubjectMapping) error {
+	if subjectMapping == nil {
+		return fmt.Errorf("subject mapping is nil: %w", ErrInvalidSubjectMapping)
+	}
+	if subjectMapping.GetAttributeValue() == nil {
+		return fmt.Errorf("subject mapping's attribute value is nil: %w", ErrInvalidSubjectMapping)
+	}
+	if subjectMapping.GetAttributeValue().GetFqn() == "" {
+		return fmt.Errorf("subject mapping's attribute value FQN is empty: %w", ErrInvalidSubjectMapping)
+	}
+	if subjectMapping.GetActions() == nil {
+		return fmt.Errorf("subject mapping's actions are nil: %w", ErrInvalidSubjectMapping)
+	}
+	return nil
+}
+
+// validateAttribute validates the attribute is valid for an entitlement decision
+//
+// attribute:
+//
+//   - must not be nil
+//   - must have a non-empty FQN
+//   - must have non-empty values
+//   - must have non-empty values FQNs
+func validateAttribute(attribute *policy.Attribute) error {
+	if attribute == nil {
+		return fmt.Errorf("attribute is nil: %w", ErrInvalidAttributeDefinition)
+	}
+	if attribute.GetFqn() == "" {
+		return fmt.Errorf("attribute FQN is empty: %w", ErrInvalidAttributeDefinition)
+	}
+	if len(attribute.GetValues()) == 0 {
+		return fmt.Errorf("attribute values are empty: %w", ErrInvalidAttributeDefinition)
+	}
+	for _, value := range attribute.GetValues() {
+		if value == nil {
+			return fmt.Errorf("attribute value is nil: %w", ErrInvalidAttributeDefinition)
+		}
+		if value.GetFqn() == "" {
```

Comment:
Should we be asserting something like:

```suggestion
		if strings.HasPrefix(value.GetFqn(), attribute.GetFqn()) {
```